src/utils/test/util.spec.ts

```diff
@@ -155,18 +155,30 @@ describe('util', () => {
       // improperly formatted JSON - note the lack of ':'
       const diagnostic = util.parsePackageJson('{ "someJson" "value"}', mockPackageJsonPath);
 
+      // due to missing colon in input
+      const messageText =
+        Number(process.versions.node.split('.')[0]) > 20
+          ? 'Unexpected string in JSON at position 13'
+          : "Expected ':' after property name in JSON at position 13";
+
       const expectedDiagnostic: d.Diagnostic = stubDiagnostic({
         absFilePath: mockPackageJsonPath,
         header: 'Error Parsing JSON',
-        messageText: 'Unexpected string in JSON at position 13', // due to missing colon in input
         type: 'build',
       });
 
-      expect(diagnostic).toEqual<util.ParsePackageJsonResult>({
-        diagnostic: expectedDiagnostic,
-        data: null,
-        filePath: mockPackageJsonPath,
-      });
+      expect(diagnostic.filePath).toBe(mockPackageJsonPath);
```

Comment:
I think we can avoid splitting all this out (and fixing the unused variable issues) if we use the [`expect.stringMatching` matcher:](https://jestjs.io/docs/expect#expectstringmatchingstring--regexp)

```
    it('returns a parse error if parsing cannot complete', () => {
      // improperly formatted JSON - note the lack of ':'
      const diagnostic = util.parsePackageJson('{ "someJson" "value"}', mockPackageJsonPath);

      const expectedDiagnostic: d.Diagnostic = stubDiagnostic({
        absFilePath: mockPackageJsonPath,
        header: 'Error Parsing JSON',
        messageText: expect.stringMatching(/.*in JSON at position 13$/),
        type: 'build',
      });

      expect(diagnostic).toEqual<util.ParsePackageJsonResult>({
        diagnostic: expectedDiagnostic,
        data: null,
        filePath: mockPackageJsonPath,
      });
    });
```
Thoughts?